keepassxc-browser/background/keepass.js

```diff
@@ -1005,3 +1005,41 @@ keepass.decrypt = function(input, nonce) {
     const res = nacl.box.open(m, n, keepass.serverPublicKey, keepass.keyPair.secretKey);
     return res;
 };
+
+keepass.enableAutomaticReconnect = function() {
+    setInterval(() => {
+        if (!keepass.isKeePassXCAvailable) {
+            keepass.connectToNative();
+            keepass.reconnect();
+        }
+    }, 1000);
+};
+
+keepass.reconnect = function(callback, tab) {
+    return new Promise((resolve, reject) => {
+        keepass.generateNewKeyPair();
+        keepass.changePublicKeys(tab).then((pkRes) => {
+            // Database hash should be received if the reconnection succeeded
+            keepass.getDatabaseHash((gdRes) => {
+                if (!gdRes) {
+                    reject(false);
```

Comment:
Do we need a return after this?

Reply:
Without a return, keepass.testAssociation will still be called after rejection and may produce ugly side effects.

Reply:
Debugged it and that indeed happens.. added the return.

Reply:
You cannot change the state of a promise once it's been rejected or resolved, but the code after the call to reject() or resolve() is still being executed.